### PR TITLE
changed: move CFileItem::GetTBNFile to ArtUtils

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Introduce a new compile unit ArtUtils and move CFileItem::GetTBNFile into it.

## Motivation and context
Move more code out of CFileItem.

## How has this been tested?
It builds + I have added a test

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by enen92 — APPROVED** [review]

**Review by garbear — APPROVED** [review]
And it includes test cases. I like it.

File: xbmc/utils/ArtUtils.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+#include <string>
+
+class CFileItem;
+
+namespace KODI::ART
+{
+
+// Gets the .tbn file associated with an item
+std::string GetTBNFile(const CFileItem& item);
```
Comment (ksooo) [inline xbmc/utils/ArtUtils.h:19]: Would it make sense to declare this function ‘static’?

**Review by ksooo — COMMENTED** [review]
Most probably you just copied most of the code to the new cpp, however it might be a good idea to adapt the copied code to our coding guidelines (with separate commit)? Example: Remove type prefix from variable names.

**Comment by notspiff** [comment]
clang format truely sucks the fun out of everything.

**Pull request merged**